### PR TITLE
add custom transforms cumsum and cumprod

### LINKED ISSUE #97: Implement JVP rules for cumsum/cumprod
Currently, cumsum/cumprod does not have jvp rule in jax yet. And these operators are necessary for simplex constraint. We should make custom primitives to support these operators.

### PR DESCRIPTION
**Pull request by fehiepsi**
Fixes #97.

### CODE REVIEW

File: test/test_distributions_util.py [hunk 1]
```diff
@@ -81,6 +81,31 @@ def test_binary_cross_entropy_with_logits(x, y):
...
+
+
+@pytest.mark.parametrize('shape', [
+    (3,),
+    (5, 3),
+])
+def test_cumprod_jac(shape):
+    rng = random.PRNGKey(0)
+    x = random.uniform(rng, shape=shape)
+    expected = grad(lambda x: np.sum(x[..., 0] + x[..., 0] * x[..., 1]
```
Comment (neerajprad) [inline test/test_distributions_util.py]: nit: Might be worth simplifying and checking the full gradient:

```python

    def test_fn(x): return np.stack([x[..., 0], x[..., 0] * x[..., 1], x[..., 0] * x[..., 1] * x[..., 2]], -1)

    assert_allclose(cumprod(x), test_fn(x))
    assert_allclose(jacfwd(cumprod)(x), jacfwd(test_fn)(x))
```

File: test/test_distributions.py [hunk 2]
```diff
@@ -393,17 +393,8 @@ def test_biject_to(constraint, shape):
...
-                expected = -1.3968685
-            elif shape == (3,):
-                expected = -7.139747
-            elif shape == (5,):
-                expected = -11.871904
-            inv_expected = -expected
-            # TODO compare against numerical values when jvp rule for `cumsum`, `cumprod` is available
-            # expected = np.linalg.slogdet(jax.jacobian(transform)(x)[::-1, :])[1]
-            # inv_expected = np.linalg.slogdet(jax.jacobian(transform.inv)(y)[:, ::-1])[1]
+            expected = np.linalg.slogdet(jax.jacobian(transform)(x)[:-1, :])[1]
```
Comment (neerajprad) [inline test/test_distributions.py]: Nice!

**Comment by fehiepsi** [comment]
@neerajprad I don’t have much  access to computer recently. Just merge this to unblock dirichlet models. I will address the test for cumsum in a later PR.
